pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -146,53 +152,88 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
+		if podReusableMemory >= requestedSize {
+			requestedSize = 0
+		} else {
+			requestedSize -= podReusableMemory
+		}
+
 		// Update nodes memory state
-		for _, nodeID := range maskBits {
-			machineState[nodeID].NumberOfAssignments++
-			machineState[nodeID].Cells = maskBits
+		p.updateMachineState(machineState, maskBits, resourceName, requestedSize)
+	}
 
-			// we need to continue to update all affinity mask nodes
-			if requestedSize == 0 {
-				continue
-			}
+	p.updatePodReusableMemory(pod, container, containerBlocks)
 
-			// update the node memory state
-			nodeResourceMemoryState := machineState[nodeID].MemoryMap[resourceName]
-			if nodeResourceMemoryState.Free <= 0 {
-				continue
-			}
+	s.SetMachineState(machineState)
+	s.SetMemoryBlocks(podUID, container.Name, containerBlocks)
 
-			// the node has enough memory to satisfy the request
-			if nodeResourceMemoryState.Free >= requestedSize {
-				nodeResourceMemoryState.Reserved += requestedSize
-				nodeResourceMemoryState.Free -= requestedSize
-				requestedSize = 0
-				continue
-			}
+	// update init containers memory blocks to reflect the fact that we re-used init containers memory
+	// it possible that the size of the init container memory block will have 0 value, when all memory
+	// allocated for it was re-used
+	p.updateInitContainersMemoryBlocks(s, pod)
+
+	return nil
+}
 
-			// the node does not have enough memory, use the node remaining memory and move to the next node
-			requestedSize -= nodeResourceMemoryState.Free
-			nodeResourceMemoryState.Reserved += nodeResourceMemoryState.Free
-			nodeResourceMemoryState.Free = 0
+func (p *staticPolicy) updateMachineState(machineState state.NUMANodeMap, numaAffinity []int, resourceName v1.ResourceName, requestedSize uint64) {
+	for _, nodeID := range numaAffinity {
+		machineState[nodeID].NumberOfAssignments++
+		machineState[nodeID].Cells = numaAffinity
+
+		// we need to continue to update all affinity mask nodes
+		if requestedSize == 0 {
+			continue
+		}
+
+		// update the node memory state
+		nodeResourceMemoryState := machineState[nodeID].MemoryMap[resourceName]
+		if nodeResourceMemoryState.Free <= 0 {
+			continue
 		}
+
+		// the node has enough memory to satisfy the request
+		if nodeResourceMemoryState.Free >= requestedSize {
+			nodeResourceMemoryState.Reserved += requestedSize
+			nodeResourceMemoryState.Free -= requestedSize
+			requestedSize = 0
+			continue
+		}
+
+		// the node does not have enough memory, use the node remaining memory and move to the next node
+		requestedSize -= nodeResourceMemoryState.Free
+		nodeResourceMemoryState.Reserved += nodeResourceMemoryState.Free
+		nodeResourceMemoryState.Free = 0
 	}
+}
 
-	s.SetMachineState(machineState)
-	s.SetMemoryBlocks(string(pod.UID), container.Name, containerBlocks)
+func (p *staticPolicy) getPodReusableMemory(pod *v1.Pod, bitMask bitmask.BitMask, resourceName v1.ResourceName) uint64 {
```

Comment:
can we call the second parameter `numaAffinity` for consistency with other places.

Reply:
done